src/lib/ppx_register_event/register_event.ml

```diff
@@ -0,0 +1,201 @@
+open Core_kernel
+open Ppxlib
+
+let deriver = "register_event"
+
+(* module in Structured_events library *)
+let module_name = "Structured_events"
+
+let hash name =
+  let open Digestif.SHA256 in
+  let ctx0 = init () in
+  let ctx1 = feed_string ctx0 name in
+  get ctx1 |> to_raw_string
+
+(* we omit an explicit `Core_kernel`, because the Ppx_deriving_yojson functions
+   generate uses of `Result` without it, so that module will need to be opened in
+   any case
+*)
+let core_kernel_result f = String.concat ["Result"; f] ~sep:"."
+
+let type_ext_str ~options ~path (ty_ext : type_extension) =
+  let add_module_qualifier s = String.concat ~sep:"." [module_name; s] in
```

Comment:
Nit: Don't think we need this abstraction, code would be clearer with `Structured_events.register_constructor` etc. directly in the metaquot sections.

Reply:
The reason I've done this is to write down the module name once only, so you don't make errors. I've been trained to do this!

Reply:
I'm not compelled by this argument, and I don't think it justifies the worse readability or additional ~5% LOC, but it's not worth blocking the PR over this.